rest/api_test.go

```diff
@@ -9719,10 +9719,10 @@ func TestSyncFnTimeout(t *testing.T) {
 	syncFnFinishedWG.Add(1)
 	go func() {
 		response := rt.SendAdminRequest("PUT", "/db/doc", `{"foo": "bar"}`)
-		requireStatus(t, response, 500) // TODO: Change to assertStatus when CBG-2143 merged maybe with response matching if response is changed
+		assertHTTPErrorReason(t, response, 500, "JS sync function timed out")
```

Comment:
I don't think this needs to hit this code review but I'd rather see `requireHTTPErrorReason`

Reply:
Having a require in the Go routine causes the test to carry on hanging until the timeout is hit on line 9725. I wanted this assert so if the test does fail, it does not hang until the timeout is hit, but rather does `syncFnFinishedWG.Done()` so the test can finish instantly.